feat: add tutorial for Swap&Bridge

shoom3301 opened the pull request:
# Description

A tutorial for Swap&Bridge feature, describes the way from token selection to bridging results tracking.

### [Live Preview](https://docs-git-feat-swap-and-bridge-tutorial-cowswap.vercel.app/cow-protocol/tutorials/cow-swap/swap-and-bridge)

<img width="1026" height="972" alt="image" src="https://github.com/user-attachments/assets/efad6c65-8290-484b-9948-1f7a6689c9a7" />


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Added a comprehensive "Swap & Bridge" tutorial guiding cross‑network swaps: setup, approvals/signing, custom recipients, quote breakdown (costs/ETA/min receive), Account Proxy behavior, one‑transaction execution flow, failure handling/refunds, and order tracking — includes step‑by‑step visuals.
  * Updated sidebar ordering for Cow Swap tutorials (flash‑loans, limit, twap, native) to improve navigation and reflect the revised tutorial sequence.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shoom3301 committed:
feat: add tutorial for Swap&Bridge

alfetopito commented on docs/cow-protocol/tutorials/cow-swap/swap-and-bridge.mdx:
```diff
@@ -0,0 +1,198 @@
+
+:::note
+
+**Tokens must be different.** Bridging the same token between two networks is not yet supported. This feature is under development.
```
To set expectations, I would not say that. Something like:

```suggestion
**Tokens must be different.** Bridging the same token between two networks is not yet supported. This feature will be available in the future.
```

alfetopito commented on docs/cow-protocol/tutorials/cow-swap/swap-and-bridge.mdx:
```diff
@@ -0,0 +1,198 @@
+* Regular ERC-20 allowance transactions
+* [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963) permit flow (when supported)
+
+<img src={SignApprovalImg.src} alt="Sign an approval" />
```
Perfect! I came here to say the same thing https://eips.ethereum.org/EIPS/eip-2612

alfetopito commented on docs/cow-protocol/tutorials/cow-swap/swap-and-bridge.mdx:
```diff
@@ -0,0 +1,198 @@
+
+### Signing Steps
+
+1. **Sign the post-hook call** for the bridge provider.
```
This section could be expanded a bit further.
Explain the order and what is being signed in the first request (the bridge calldata).

shoom3301 replied:
Thanks!
Added more details:

<img width="995" height="345" alt="image" src="https://github.com/user-attachments/assets/1ae9ce6f-d643-442e-b9ba-c6890c82a98f" />

alfetopito reviewed (commented):
Nice work!
Added a few suggestions.

elena-zh committed:
Update swap-and-bridge.mdx

elena-zh committed:
Update swap-and-bridge.mdx

elena-zh committed:
Update swap-and-bridge.mdx

shoom3301 committed:
chore: fix typos

elena-zh commented:
Hey @shoom3301 , great job, thank you!
I've proposed some changes (feel free to revert them if they are not relevant) :) 

And there are still some questions/proposals from my side: 
1. Should we mention something about (soft) cancelling an order that it is possible, but only in the swap part? And that the bridge TX is irreversible unless it fails? 
2. I'd add an anchor link to the eponymous section for **Account Proxy** 
<img width="500" height="169" alt="image" src="https://github.com/user-attachments/assets/621a0ed4-3d81-45c7-880c-52de8c74f149" />
<img width="1043" height="263" alt="image" src="https://github.com/user-attachments/assets/e7c0a565-71d2-4b9d-abdf-831dd8e98fc4" />
3. I'd add a comment that only last 10 orders are displayed in the Account modal. All the rest orders can be found on Explorer
<img width="1013" height="690" alt="image" src="https://github.com/user-attachments/assets/dcfe1c1c-07ab-4dad-a6e2-2518811244d3" />

Thanks!

shoom3301 commented on docs/cow-protocol/tutorials/cow-swap/swap-and-bridge.mdx:
```diff
@@ -0,0 +1,210 @@
+
+<img src={SwapSuccessImg.src} style={{ maxWidth: 500 }} alt="Swap is finished" />
+
+>If your order expires, is canceled, or fails, bridging will not start — your funds will remain in your wallet.
```
Nice!

shoom3301 committed:
chore: update s&b docs

shoom3301 commented:
@elena-zh thank you!

1. Added info about cancelling as [Rabbit suggested](https://github.com/cowprotocol/docs/pull/526#discussion_r2266483594).
2. Fixed
3. Fixed

elena-zh commented:
Hey @shoom3301 , could you please check the latest deployment? I don't see the latest changes there

shoom3301 committed:
chore: fix build

shoom3301 commented:
@elena-zh sorry, fixed

elena-zh committed:
add a note

elena-zh committed:
added details to p.7-8

elena-zh commented:
Hey @shoom3301 , thank you!
I've also added some improvements to the doc. Please, review them and LMK if they are OK for you.

shoom3301 committed:
chore: add info about bridge post-hook

shoom3301 committed:
Merge remote-tracking branch 'origin/feat/swap-and-bridge-tutorial' into feat/swap-and-bridge-tutorial

pretf00d reviewed (commented):
Hey Sasha, just one thing, we're officially calling these functionality Cross-chain swaps. Can you update the language to reflect that? then I can get it approved for you.

shoom3301 committed:
chore: rename Cross-chain swaps

shoom3301 commented:
> Hey Sasha, just one thing, we're officially calling these functionality Cross-chain swaps. Can you update the language to reflect that? then I can get it approved for you.

Thanks! Renamed it

pretf00d commented on docs/cow-protocol/tutorials/cow-swap/swap-and-bridge.mdx:
```diff
@@ -0,0 +1,239 @@
+
+## 4. Previewing Your Quote
+
+In the Swap form you can see a detailed breakdown of the expected swap and bridge:
```
cross-chain swap.

shoom3301 replied:
In that case, it is correct, because we are talking about swap and then bridge details:

<img width="497" height="513" alt="image" src="https://github.com/user-attachments/assets/1c317b06-0dfd-40d4-8df8-37d2f1f2015c" />

pretf00d reviewed (commented):
Just one small comment from my side.
